tests/factories.py

```diff
@@ -489,27 +477,89 @@ def review_file(request, relpath, status, *users):
         # use the DAL directly means we can add reviews regardless of request
         # state, which is very useful in test setup.
         if status == RequestFileVote.APPROVED:
-            bll._dal.approve_file(
+            bll.approve_file(
                 request,
-                relpath=relpath,
-                username=user.username,
-                audit=create_audit_event(
-                    AuditEventType.REQUEST_FILE_APPROVE, user=user.username
-                ),
+                request.get_request_file_from_output_path(relpath),
+                user=user,
             )
         elif status == RequestFileVote.REJECTED:
-            bll._dal.reject_file(
+            bll.reject_file(
                 request,
-                relpath=relpath,
-                username=user.username,
-                audit=create_audit_event(
-                    AuditEventType.REQUEST_FILE_REJECT, user=user.username
-                ),
+                request.get_request_file_from_output_path(relpath),
+                user=user,
             )
         else:
             raise AssertionError(f"unrecognised status; {status}")  # pragma: no cover
 
 
+@dataclass
+class TestRequestFile:
+    """Placeholder containing file metadata.
+
+    Allows us to set up file states declaratively. Use the add() and vote()
+    methods can be called with a request in the right state.
```

Comment:
this reads like it's been edited weirdly
```suggestion
    Allows us to set up file states declaratively. The add() and vote()
    methods can be called with a request in the right state.
```